cashu/wallet/wallet.py

```diff
@@ -201,18 +264,27 @@ async def pay_lightning(self, proofs: List[Proof], amount: int, invoice: str):
         return status["paid"]
 
     @staticmethod
-    async def serialize_proofs(proofs: List[Proof]):
-        proofs_serialized = [p.to_dict() for p in proofs]
+    async def serialize_proofs(proofs: List[Proof], hide_secrets=False):
+        if hide_secrets:
+            proofs_serialized = [p.to_dict_no_secret() for p in proofs]
+        else:
+            proofs_serialized = [p.to_dict() for p in proofs]

```

Comment:
If secret was used (via `cashu send <amount> -s <secret>`), do not serialize secret in token that is going to be shared (i.e., do not reveal the "secret").